io/disk/htx_block_devices.py

```diff
@@ -0,0 +1,178 @@
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Author: Naresh Bannoth<nbannoth@in.ibm.com>
+# this script run IO stress on block devices for give time.
+
+import os
+# import sys
+import time
+from avocado import Test
+from avocado import main
+# from commands import *
+from avocado.utils.software_manager import SoftwareManager
+# from avocado.utils import archive
+# from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import distro
+
+
+class HtxTest(Test):
+
+    """
+    HTX [Hardware Test eXecutive] is a test tool suite. The goal of HTX is to
+    stress test the system by exercising all hardware components concurrently
+    in order to uncover any hardware design flaws and hardware hardware or
+    hardware-software interaction issues.
+    :see:https://github.com/open-power/HTX.git
+    :param block_devices: names of block_devices on which you want to run HTX
+    :param mdt_file: mdt file used to trigger HTX
+    :params time_limit: how much time you want to run this stress.
+    """
+
+    def setUp(self):
+        """
+        Build 'HTX'.
+        """
+        self.distro_name = distro.detect().name
+        packages = ['git', 'gcc', 'make', 'libncurses5', 'g++', 'libdapl-dev',
+                    'ncurses-dev', 'libncurses-dev']
+        smm = SoftwareManager()
+        if self.distro_name == 'Ubuntu':
+            for pkg in packages:
+                if not smm.check_installed(pkg) and not smm.install(pkg):
+                    self.skip("Can not install %s" % pkg)
+        else:
+            self.skip("skipping for other than Ubuntu Distro")
+
+        cmd = "git clone https://www.github.com/open-power/HTX"
+        os.chdir(self.teststmpdir)
+        process.system(cmd)
+        os.chdir('HTX')
+        process.run('make all')
+        process.run('make deb')
+        process.run('dpkg -r htxubuntu')
+        process.run('dpkg --purge htxubuntu')
+        process.run('dpkg -i htxubuntu.deb')
+        self.mdt_file = self.params.get('mdt_file', default='mdt.hd')
+        self.time_limit = self.params.get('time_limit', default=600)
+        self.block_device = self.params.get('block_devices', default=None)
+        if self.block_device is None:
+            self.skip("Needs the block devices to run the HTX")
+
+    def test(self):
+        """
+        Execute 'HTX' with appropriate parameters.
+        """
+        self.log.info("Starting the HTX Deamon")
+        process.run('/usr/lpp/htx/etc/scripts/htxd_run')
+
+        self.log.info("selecting the mdt file ")
+        cmd = "htxcmdline -select -mdt %s" % self.mdt_file
+        process.system(cmd)
```

Comment:
please use  `ignore_status=True ` as param in  `process.system` as this command error out script also error out